src/main/java/com/synopsys/integration/jenkins/detect/service/strategy/DetectStrategyService.java

```diff
@@ -34,18 +44,36 @@
     private final JenkinsIntLogger logger;
     private final String remoteTempWorkspacePath;
     private final JenkinsProxyHelper jenkinsProxyHelper;
+    private final JenkinsConfigService jenkinsConfigService;
 
-    public DetectStrategyService(JenkinsIntLogger logger, JenkinsProxyHelper jenkinsProxyHelper, String remoteTempWorkspacePath) {
+    public DetectStrategyService(JenkinsIntLogger logger, JenkinsProxyHelper jenkinsProxyHelper, String remoteTempWorkspacePath, JenkinsConfigService jenkinsConfigService) {
         this.logger = logger;
         this.jenkinsProxyHelper = jenkinsProxyHelper;
         this.remoteTempWorkspacePath = remoteTempWorkspacePath;
+        this.jenkinsConfigService = jenkinsConfigService;
     }
 
-    public DetectExecutionStrategy getExecutionStrategy(IntEnvironmentVariables intEnvironmentVariables, OperatingSystemType operatingSystemType, String remoteJdkHome) {
-        DetectExecutionStrategy detectExecutionStrategy;
+    public DetectExecutionStrategy getExecutionStrategy(IntEnvironmentVariables intEnvironmentVariables, OperatingSystemType operatingSystemType, String remoteJdkHome, DetectDownloadStrategy detectDownloadStrategy)
+        throws IOException, InterruptedException, DetectJenkinsException {
+        if (detectDownloadStrategy == null || detectDownloadStrategy instanceof InheritFromGlobalDownloadStrategy) {
+            Optional<DetectGlobalConfig> globalConfiguration = jenkinsConfigService.getGlobalConfiguration(DetectGlobalConfig.class);
+
+            if (!globalConfiguration.isPresent()) {
+                throw new DetectJenkinsException("Unable to identify Detect Global Configuration.");
+            }
+
+            detectDownloadStrategy = globalConfiguration.get().getDownloadStrategy();
+        }
+
+        logger.info("Running Detect using strategy: " + detectDownloadStrategy);
+
         String detectJarPath = intEnvironmentVariables.getValue(DetectJenkinsEnvironmentVariable.USER_PROVIDED_JAR_PATH.stringValue());
+        DetectExecutionStrategy detectExecutionStrategy;
 
-        if (StringUtils.isNotBlank(detectJarPath)) {
+        if (detectDownloadStrategy instanceof AirGapDownloadStrategy) {
+            String airGapJarPath = new FindOrInstallAirGapJar(logger, jenkinsConfigService).getOrDownloadAirGapJar(detectDownloadStrategy);
```

Comment:
This setup work is usually done in the SetupCallables that each ExecutionStrategy provides-- can we leverage them here?